template/gitlab-runner.tftpl

```diff
@@ -183,10 +197,38 @@ EOF
 
 cat <<EOF > /opt/remove_gitlab_registration.sh
 #!/bin/bash
-echo "Removing Gitlab Runner ..."
-aws ssm put-parameter --overwrite --type SecureString  --name "${secure_parameter_store_runner_token_key}" --region "${secure_parameter_store_region}" --value="null" 2>&1
-curl -sS ${curl_cacert} --request DELETE "${runners_gitlab_url}/api/v4/runners" --form "token=$token" 2>&1
+json_token=$(aws ssm get-parameters --names "${secure_parameter_store_runner_token_key}" --with-decryption --region "${secure_parameter_store_region}" | jq -r ".Parameters | .[0] | .Value")
+deregister_runner=true
+
+# for  module versions >7.4.1 the token is JSON
+if [[ "$json_token" =~ ^\{.* ]]; then
+  usage_counter=$(echo $json_token | jq -r .usage_counter)
+
+  # ensure that the token is not in use by another Runner
+  if [[ $usage_counter -gt 1 ]]; then
+    deregister_runner=false
+    token="not needed"
+  else
+    token=$(echo $json_token | jq -r .token)
+  fi
+else
+  token=$json_token
+fi
+
+if [[ $deregister_runner == "true" ]]; then
+  echo "Removing Gitlab Runner ..."
 
+  aws ssm put-parameter --overwrite --type SecureString  --name "${secure_parameter_store_runner_token_key}" --region "${secure_parameter_store_region}" --value="{\"token\": \"null\", \"in_use_counter\": 0}" 2>&1
+  curl -sS ${curl_cacert} --request DELETE "${runners_gitlab_url}/api/v4/runners" --form "token=$token" 2>&1
+else
+  usage_counter=$(echo $json_token | jq -r .usage_counter)
+  usage_counter=$(($usage_counter-1))
+  json_token=$(echo $json_token | jq ".usage_counter = $usage_count er")
```

Comment:
I tried this jq expression locally but couldn't get it to work. I'm not familiar with the `er` part, what does it do?

Reply:
Well, it does nothing. Simply wrong. Thanks for pointing this out. Curious, I tested it on a Runner instance. Maybe the space came in by accident.

Reply:
Thanks for fixing.